libs/stats/odc/stats/tasks.py

```diff
@@ -55,16 +58,35 @@ def sanitize(v):
         if isinstance(v, datetime):
             return v.isoformat()
         return v
+
     return transform_object_tree(sanitize, query)
 
 
+def download_from_s3(cache):
+    s3 = boto3.resource("s3")
+    path_parts = cache.replace("s3://", "").split("/")
+    bucket = path_parts.pop(0)
+    key = "/".join(path_parts)
+    try:
+        filename = Path(key).name
```

Comment:
Why are we making it a Path and getting the name?

Reply:
Oh, I see, that's the local file... It's getting plonked in a path relative to wherever this thing is run.

We should either place it in a known location, like `/tmp/{key}` or in a magic tempfile (tempfile.givemeone or whatever).

Also, we should perhaps test if it exists locally first, rather than overwriting it every time we look for it.

Reply:
Please move to

`odc/aws/__init__.py` and use this function for implementation that is compatible with other s3 methods

https://github.com/opendatacube/odc-tools/blob/86dc6d37e388d6d4e1aad4603fe07a3a4ae61dcf/libs/aws/odc/aws/__init__.py#L295-L305

Reply:
I've moved this to odc/aws/__init__.py.  Reading from s3 is slow so I didn't use this function.  please check the new implementation.